src/middleware/usbboot.rs

```diff
@@ -98,9 +128,9 @@ fn map_to_serial<T: UsbContext>(dev: &rusb::Device<T>) -> anyhow::Result<String>
         .context("error reading serial")
 }
 
-pub(crate) fn verify_one_device<T>(devices: &[T]) -> std::result::Result<(), FlashingError> {
+pub(crate) fn verify_one_device<T>(devices: &[T]) -> std::result::Result<&T, FlashingError> {
```

Comment:
You can turn this into something like `extract_one_device`, akin to [`get_device_path`](https://github.com/turing-machines/tpi_rs/pull/24/files#diff-b1e54e1cb890e7aa83c18a19052af028d97a8549ef9c6add19ac19edd2028e3dR106). The original intent was a sanity check, but is now changed.

Also, all `std::result::Result`s can be turned into `anyhow::Result`s (or better just `Result`) since `FlashingError` implements `std::error::Error`.